Assets/MixedRealityToolkit.Services/InputSystem/InputSystemGlobalHandlerListener.cs

```diff
@@ -0,0 +1,164 @@
+        private readonly MethodInfo registerHandlerMethod = typeof(IMixedRealityEventSystem).GetMethod("RegisterHandler");
+        private readonly MethodInfo unregisterHandlerMethod = typeof(IMixedRealityEventSystem).GetMethod("UnregisterHandler");
+
+        private IMixedRealityInputSystem inputSystem = null;
+
+        /// <summary>
+        /// The active instance of the input system.
+        /// </summary>
+        protected IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        protected virtual void OnEnable()
+        {
+            if (InputSystem != null && !lateInitialize)
+            {
+                RegisterHandlers();
+            }
+        }
+
+        protected virtual async void Start()
+        {
+            if (lateInitialize)
+            {
+                await EnsureInputSystemValid();
+
+                // We've been destroyed during the await.
+                if (this == null)
+                {
+                    return;
+                }
+
+                lateInitialize = false;
+                RegisterHandlers();
+            }
+        }
+
+        protected virtual void OnDisable()
+        {
+            UnregisterHandlers();
+        }
+
+        /// <summary>
+        /// A task that will only complete when the input system has in a valid state.
+        /// </summary>
+        /// <remarks>
+        /// It's possible for this object to have been destroyed after the await, which
+        /// implies that callers should check that this != null after awaiting this task.
+        /// </remarks>
+        protected async Task EnsureInputSystemValid()
+        {
+            if (InputSystem == null)
+            {
+                await new WaitUntil(() => InputSystem != null);
+            }
+        }
+
+        /// <summary>
+        /// Overload this method to specify, which global events component wants to listen to.
+        /// Use RegisterHandler API of InputSystem
+        /// </summary>
+        protected abstract void RegisterHandlers();
+
+        /// <summary>
+        /// Overload this method to specify, which global events component should stop listening to.
+        /// Use UnregisterHandler API of InputSystem
+        /// </summary>
+        protected abstract void UnregisterHandlers();
+
+        /// <summary>
+        /// Utility method, which registers all handlers implemented by 'this' using Reflection.
+        /// T should be a derived class type
+        /// Avoid using this method unless class implements so many interfaces that enumerating them manually becomes unclear for the reader
+        /// </summary>
+        protected void RegisterAllHandlers<T>() where T: class, IEventSystemHandler
```

Comment:
I don't think it's worth having this utility. For me what the user gains is not worth the added maintenance burden.

Reply:
Maybe you're right. It's only used in GGVPointer implementation now. One place is not worth it.

Reply:
Removed both. Don't know why I thought adding them was a good idea in the first place. I guess I just wanted to call a generic method via reflection.

Reply:
I totally get that